nixos/tests/galene.nix

```diff
@@ -208,4 +208,136 @@ in
       '';
     }
   );
+
+  stream = makeTest (
+    { pkgs, lib, ... }:
+    let
+      galeneTestGroupBotName = "bot";
+      galeneTestGroupBotPassword = "1234";
+      galeneStreamSrtPort = 9710;
+      galeneStreamFeedImage = "galene-stream-feed.png";
+      galeneStreamFeedLabel = "Example";
+    in
+    {
+      name = "galene-stream-works";
+      meta = {
+        # inherit (pkgs.galene-stream.meta) teams;
+        inherit (pkgs.galene.meta) maintainers;
+        platforms = lib.platforms.linux;
+      };
+
+      nodes.machine =
+        { config, pkgs, ... }:
+        {
+          imports = [ ./common/x11.nix ];
+
+          services.xserver.enable = true;
+
+          environment = {
+            # https://galene.org/INSTALL.html
+            etc = {
+              ${galeneTestGroupFile}.source = (pkgs.formats.json { }).generate galeneTestGroupFile {
+                presenter = [
+                  {
+                    username = galeneTestGroupBotName;
+                    password = galeneTestGroupBotPassword;
+                  }
+                ];
+                other = [ { } ];
+              };
+
+              ${galeneStreamFeedImage}.source =
+                pkgs.runCommand galeneStreamFeedImage
+                  {
+                    nativeBuildInputs = with pkgs; [
+                      (imagemagick.override { ghostscriptSupport = true; }) # Add text to image
+                    ];
+                  }
+                  ''
+                    magick -size 400x400 -background white -fill black canvas:white -pointsize 70 -annotate +100+200 '${galeneStreamFeedLabel}' $out
+                  '';
+            };
+
+            systemPackages = with pkgs; [
+              ffmpeg
+              firefox
+              galene-stream
+            ];
+          };
+
+          services.galene = {
+            enable = true;
+            insecure = true;
+            httpPort = galenePort;
+            groupsDir = galeneTestGroupsDir;
+          };
+        };
+
+      enableOCR = true;
+
+      testScript = ''
+        machine.wait_for_x()
+
+        with subtest("galene starts"):
+            # Starts?
+            machine.wait_for_unit("galene")
+
+            # Keeps running after startup?
+            machine.sleep(10)
+            machine.wait_for_unit("galene")
```

Comment:
Can't we just wait for the port to be open instead of sleeping? Else, are there any scenarios where the service would stop running?

Reply:
I was *pretty* sure that I had some setup issues early on where Galene would start up fine, but crash shortly afterwards on something, but I can't figure out what the issue was… And we *do* check that it responds properly later on, so I guess this isn't really necessary. Will drop.